### ISSUE TITLE
[Bug]: Locale Not Rendering Correctly for Bluesky

### ISSUE THREAD

**Issue by aaronvegh** [MasterJ93/ATProtoKit#82]
### Summary

The createPostRecord API takes an array of Locale objects. When it sends to Bsky, those Locales are converted to identifiers that are not compatble with the expected data.

### Reproduction Steps

The documentation for `createPostRecord` provides the example to use:

```
let postResult = try await atProtoBluesky.createPostRecord(
      text: "Première promenade d'automne 🍂. Les couleurs sont magnifiques cette année!",
      locales: [Locale(identifier: "fr_FR")]
)
```

If I try this myself, I get the following error back from bsky:

```
ATProtoKit.ATAPIError.badRequest(error: ATProtoKit.ATHTTPResponseError(error: "InvalidRequest", message: "Invalid app.bsky.feed.post record: Record/langs/0 must be a well-formed BCP 47 language tag"))
```

It appears that a [BCP 47 language tag](https://en.wikipedia.org/wiki/IETF_language_tag#List_of_common_primary_language_subtags) is for _just_ the language code, not the region as well. So "fr_FR" represents the French language spoken in France, but only "fr" is a valid tag.

### Expected Results

I'd like to be able to provide a Locale object to the API and have it give bsky only the correct language code.

### Actual Results

Well, you saw the error. 

### What operating systems did you experience this bug?

macOS, iOS/iPadOS

### Operating System Version

macOS 15.2

### ATProtoKit Version

0.21.3

**Comment by aaronvegh** [comment]
I'll provide a PR to resolve this one shortly.

**Comment by MasterJ93** [comment]
Thanks for this. I'll approve of it shortly and release it by the end of the week (I need to see if there are any additional fixes that need to be resolved and combine them into one update).

**Comment by MasterJ93** [comment]
There was a delay (I didn't want to release the update with _just_ that bug fix alone), but the update that contains your code has been released as of now. However, it appears that it caused tvOS to have a build failure. So I've tweaked it to make sure the code is available for tvOS 16 and higher (as well as macOS 13 and watchOS 9).

(I forgot about visionOS. I'll add that in the next patch release.)

I'll be closing this issue now, since it's been resolved.
